File: src/main/java/com/ratepay/bugtracker/model/Ticket.java
package com.ratepay.bugtracker.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "ticket")
public class Ticket {
    @Id
    @GeneratedValue(strategy= GenerationType.SEQUENCE)
    @Column(name = "ticket_id")
    private long ticketId;
    private int ticketNumber;
    private String project;
    private String issueType;
    private String summary;
    private String description;
    private int severity;
    private String reporter;
}

File: src/web/src/app/model/ticket.ts
export class Ticket{
  ticketId?: number;
  ticketNumber?: number;
  project?: string;
  issueType?: string;
  summary?: string;
  description?: string;
  severity?: number;
  reporter?: string;

  constructor(
    ticketId?: number,
    ticketNumber?: number,
    project?: string,
    issueType?: string,
    description?: string,
    summary?: string,
    severity?: number,
    reporter?: string,
  ) {
      this.ticketId = ticketId,
      this.ticketNumber = ticketNumber,
      this.project = project,
      this.issueType = issueType,
      this.description = description,
      this.summary = summary,
      this.severity = severity,
      this.reporter = reporter
    }
}


